src/world/Server/Packets/Handlers/TradeHandler.cpp

```diff
@@ -394,11 +394,21 @@ void WorldSession::handleAcceptTrade(WorldPacket& /*recvPacket*/)
         {
             tradeItems[i]->setGiftCreatorGuid(_player->getGuid());
             _player->getItemInterface()->SafeRemoveAndRetreiveItemByGuid(tradeItems[i]->getGuid(), true);
+
+#if VERSION_STRING >= WotLK
+            if (tradeItems[i]->hasFlags(ITEM_FLAG_BOP_TRADEABLE))
+                tradeTarget->removeTradeableItem(tradeItems[i]);
```

Comment:
Should this be "_player->removeTradeableItem" ? Since tradeItems[] are owned by _player

Reply:
this is correct how it is